integration/Dockerfile

```diff
@@ -0,0 +1,133 @@
+ARG CONTRACTS_VERSION="0.36.0"
+ARG PFS_VERSION="918ee6ef2dcfe293d1ce57c11f5ca23c8e9d68f3"
+ARG RAIDEN_VERSION="2e741dfdf4bfa564dec760abd5e3d8b2c9d30715"
+ARG OS_NAME="LINUX"
+ARG GETH_VERSION="1.9.9"
+ARG GETH_URL_LINUX="https://gethstore.blob.core.windows.net/builds/geth-linux-amd64-1.9.9-01744997.tar.gz"
+ARG GETH_MD5_LINUX="de1b31439dc6697fcbe2d729d2080d22"
+
+FROM python:3.7 as raiden-builder
+ARG RAIDEN_VERSION
+
+# clone raiden repo + install dependencies
+RUN git clone -b develop https://github.com/raiden-network/raiden /app/raiden
+RUN python3 -m venv /opt/raiden
+ENV PATH="/opt/raiden/bin:$PATH"
+
+WORKDIR /app/raiden
+RUN git checkout ${RAIDEN_VERSION}
+RUN make install
+
+FROM python:3.7 as synapse-builder
+ARG SYNAPSE_VERSION=1.5.1
+
+RUN \
+    python -m venv /synapse-venv && \
+    /synapse-venv/bin/pip install "matrix-synapse[sqlite]==${SYNAPSE_VERSION}"
+
+RUN /synapse-venv/bin/pip install psycopg2 coincurve pycryptodome
+
+COPY synapse/eth_auth_provider.py /synapse-venv/lib/python3.7/site-packages/
+COPY synapse/admin_user_auth_provider.py /synapse-venv/lib/python3.7/site-packages/
+
+FROM python:3.7
+LABEL maintainer="Raiden Network Team <contact@raiden.network>"
+
+ARG OS_NAME
+ARG GETH_URL_LINUX
+ARG GETH_MD5_LINUX
+ARG CONTRACTS_VERSION
+ARG GETH_VERSION
+
+RUN apt-get update && apt-get install -y supervisor --no-install-recommends python3-virtualenv python3-dev
+
+ENV PASSWORD_FILE=/opt/passwd
+ENV PASSWORD=1234
+RUN echo ${PASSWORD} > ${PASSWORD_FILE}
+
+# prepare raiden
+COPY --from=raiden-builder /opt/raiden /opt/raiden
+RUN mkdir -p /opt/raiden/config/keys
+COPY geth/keystores/* /opt/raiden/config/keys/
+COPY raiden/* /opt/raiden/config/
+
+# Download GETH
+ARG DEPLOYMENT_DIRECTORY=/opt/deployment
+ARG VENV=/opt/raiden
+ARG SMARTCONTRACTS_ENV_FILE=/etc/profile.d/smartcontracts.sh
+
+COPY geth/download_geth.sh /usr/local/bin
+COPY geth/genesis.py /usr/local/bin
+
+ARG LOCAL_BASE=/usr/local
+RUN download_geth.sh
+
+COPY geth/deploy_contracts.py /usr/local/bin
+COPY geth/deploy.sh /usr/local/bin
+
+ARG DATA_DIR=/opt/chain
+RUN deploy.sh
```

Comment:
Consecutive RUNs can/should be concatenated together with `&& \` in order to reduce the number of intermediary images: docker keeps each command on a Dockerfile as a separate image on disk (although deduplicated with the others), and this can save up several GBs on large Dockerfiles.
Also, any in-disk files downloaded/created on intermediary images are kept because of this, that's why you usually see `apt-get update && ... && rm -rf /var/...` RUN commands, so these apt-get files/cache/dbs aren't kept on these intermediary images

Reply:
I can try :) the whole final image at this point is around 1.6GiB though (after removing ethash)

Reply:
Any tips you have on how to improve though are really welcome.

Reply:
Yes, this shouldn't help much with the final image size, but can at least save a lot in the intermediary ones. My main tip is, try to group all similar commands (as much as possible, of course, as some may depend on others) and check if there're caches left behind that could/should be removed (e.g. pip caches, npm caches, apt caches, etc)

Reply:
We should not have any npm caches in the image but I am looking into the other :) thanks

Reply:
But I can see that the install I run is actually responsible for a great part of the size. That steps add around 500MB. I am trying to figure out if somehow unneeded dependencies are installed.